src/CoinParamUtils.cpp

```diff
@@ -129,6 +130,129 @@ std::string nextField(const char *prompt)
 
 namespace CoinParamUtils {
 
+std::string printString(const std::string input, int maxWidth)
+{
+   std::ostringstream buffer;
+   std::istringstream iss(input);
+   int currentWidth = 0;
+   std::string word;
+   while (iss >> word){
+      currentWidth += word.length();
+      if (currentWidth >= 65){
+         buffer << std::endl << word;
+      } else {
+         buffer << word;
+      }         
+   }
+   return buffer.str();
+}
+
+std::string printString(const char *input, int maxWidth)
+{
+   return CoinParamUtils::printString(std::string(input));
+}
+
+void
+readFromStream(std::deque<std::string> &inputQueue,
+               std::istream &inputStream)
+{
+   std::string field;
+   while (inputStream >> field){
+      std::string::size_type found = field.find('=');
+      if (found != std::string::npos) {
+         inputQueue.push_back(field.substr(0, found));
+         inputQueue.push_back(field.substr(found + 1));
+      } else {
+         inputQueue.push_back(field);
+      }
+   }
+}
+
+void
+readInteractiveInput(std::deque<std::string> &inputQueue,
+                     std::string prompt)
+{
+   std::string input;
+  
+#ifdef COINUTILS_HAS_READLINE
+   // Get a line from the user.
+   input = std::string(readline(prompt.c_str()));
+   // If the line has any text in it, save it on the history.
+   if (input.length() > 0) {
+      add_history(input.c_str());
+   }
+#else
+   std::cout << prompt;
+   fflush(stdout);
+   getline(std::cin, input); 
+#endif
+   if (!input.length()){
+      return; 
+   }
+
+   std::istringstream inputStream(input);
+   CoinParamUtils::readFromStream(inputQueue, inputStream);
+}
+
+std::string
+getNextField(std::deque<std::string> &inputQueue, bool interactiveMode,
+             std::string prompt)
+{
+  if (inputQueue.empty() && interactiveMode){
+     CoinParamUtils::readInteractiveInput(inputQueue, prompt);
+  }
+
+  if (inputQueue.empty()){
+     return "";
+  }else{
+     std::string field = inputQueue.front();
+     inputQueue.pop_back();
+     return field;
+  }
+}
+
+int getValue(std::deque<std::string> &inputQueue, std::string &value)
+{
+   value = CoinParamUtils::getNextField(inputQueue);
+
+   if (value == "--" || value == "stdin"){
+      value == "-";
+   } else if (value == "stdin_lp"){
+      value == "-lp";
+   }
+
+   return value == "" ? 1 : 0;
+}
+
+// return 0 - okay, 1 bad, 2 not there
+int getValue(std::deque<std::string> &inputQueue, int &value)
+{
+   std::string field = CoinParamUtils::getNextField(inputQueue);
+
+   if (field.empty()){
+      return 2;
+   }
+   
+   char c;
+   std::stringstream ss(field);
+   ss >> value;
```

Comment:
if we can't use std::stoX due to c++98, can we at least kill this copy&paste by using a template (or redirecting to a "hidden" template)?

Reply:
Can you clarify this one? Do you mean the fact that this code snippet is repeated? I'm open to alternatives to anything here, I wouldn't exactly say I'm an expert on best practices in C++.